internal/providers/pke/pkeworkflow/update_cluster.go

```diff
@@ -84,52 +83,80 @@ func UpdateClusterWorkflow(ctx workflow.Context, input UpdateClusterWorkflowInpu
 	}
 
 	// delete removed nodepools
+	var caughtErrors []error
+	futures := make(map[string]workflow.Future, len(input.NodePoolsToDelete))
 	for _, np := range input.NodePoolsToDelete {
 		if np.Master || !np.Worker {
 			continue
 		}
 
-		err := workflow.ExecuteActivity(ctx, DeletePoolActivityName, DeletePoolActivityInput{
+		futures[np.Name] = workflow.ExecuteActivity(ctx, DeletePoolActivityName, DeletePoolActivityInput{
 			// AWSActivityInput: awsActivityInput,
 			ClusterID: input.ClusterID,
 			Pool:      np,
-		}).Get(ctx, nil)
-		if err != nil {
-			return emperror.Wrapf(err, "deleting %q", np.Name)
+		})
+	}
+	for name, future := range futures {
+		if err := future.Get(ctx, nil); err != nil {
+			caughtErrors = append(caughtErrors, errors.Wrapf(err, "couldn't delete nodepool %q", name))
 		}
 	}
+	if len(caughtErrors) != 0 {
+		return errors.Combine(caughtErrors...)
+	}
 
 	// create/change nodepools that are not removed
+	futures = make(map[string]workflow.Future, len(input.NodePoolsToUpdate))
 	for _, np := range input.NodePoolsToUpdate {
 		stackName := fmt.Sprintf("pke-pool-%s-worker-%s", input.ClusterName, np.Name)
-		var cfOut map[string]string
 
-		err := workflow.ExecuteActivity(ctx,
+		futures[np.Name] = workflow.ExecuteActivity(ctx,
 			WaitCFCompletionActivityName,
 			WaitCFCompletionActivityInput{
 				AWSActivityInput: awsActivityInput,
-				StackID:          stackName}).Get(ctx, &cfOut)
-		if err != nil {
-			return emperror.Wrap(err, fmt.Sprintf("can't find AutoScalingGroup for pool %q", np.Name))
+				StackID:          stackName})
+	}
+
+	caughtErrors = nil
```

Comment:
no need to assign `nil` because if we got here `caughtErrors` will still be `nil`